linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/ResponseClassifiers.scala

```diff
@@ -0,0 +1,124 @@
+package io.buoyant.linkerd.protocol.http
+
+import com.twitter.finagle.http.{Method, Request, Response}
+import com.twitter.finagle.service.{ResponseClass, ReqRep, ResponseClassifier}
+import com.twitter.util.{NonFatal, Return, Throw}
+import io.buoyant.config.ConfigInitializer
+import io.buoyant.linkerd.{ResponseClassifierConfig, ResponseClassifierInitializer}
+
+object ResponseClassifiers {
+
+  object Requests {
+
+    case class ByMethod(methods: Set[Method]) {
+      def unapply(req: Request): Boolean = methods.contains(req.method)
+    }
+
+    /** Matches read-only requests */
+    val ReadOnly = ByMethod(Set(
+      Method.Get,
+      Method.Head,
+      Method.Options,
+      Method.Trace
+    ))
+
+    /**
+     * Matches idempotent requests.
+     *
+     * Per RFC2616:
+     *
+     *   Methods can also have the property of "idempotence" in that
+     *   (aside from error or expiration issues) the side-effects of N >
+     *   0 identical requests is the same as for a single request. The
+     *   methods GET, HEAD, PUT and DELETE share this property. Also,
+     *   the methods OPTIONS and TRACE SHOULD NOT have side effects, and
+     *   so are inherently idempotent.
+     */
+    val Idempotent = ByMethod(Set(
+      Method.Get,
+      Method.Head,
+      Method.Put,
+      Method.Delete,
+      Method.Options,
+      Method.Trace
+    ))
+  }
+
+  object Responses {
+
+    object Failure {
+      def unapply(rsp: Response): Boolean =
+        rsp.statusCode >= 500 && rsp.statusCode <= 599
```

Comment:
finagle-http provides very similar functionality in `Status.ServerError.unapply`, I think.